lib/chore_runner/input.ex

```diff
@@ -1,13 +1,14 @@
 defmodule ChoreRunner.Input do
-  @valid_types ~w(string int float file bool)a
+  @valid_types ~w(string int float file bool select)a
 
-  @type input_type :: :string | :int | :float | :file | :bool
+  @type input_type :: :string | :int | :float | :file | :bool | :select
   @type reason :: atom() | String.t()
   @type validator_function ::
           (term() -> {:ok, term()} | :ok | true | {:error, reason} | nil | false)
   @type input_options :: [
           validators: [validator_function],
-          description: String.t()
+          description: String.t(),
+          options: Keyword.t()
```

Comment:
I think it's Ok to break the mold here and make select a special 3 arity input function. For now, I would do it like so:
```elixir
  for type <- @valid_types do
    if type == :select do
      @spec unquote(type)(atom(), Keyword.t(), input_options) :: t
      def select(name, select_options, opts \\ []) do
        {:select, name, select_options, opts}
      end
    else
      @spec unquote(type)(atom(), input_options) :: t
      def unquote(type)(name, opts \\ []) do
        {unquote(type), name, opts}
      end
    end
  end
```

Reply:
Will need to modify the input template as well to expect this special case